gitbutler-app/src/virtual_branches/integration.rs

```diff
@@ -319,14 +318,8 @@ fn verify_head_is_set(
         Some(refname) if refname.to_string() == GITBUTLER_INTEGRATION_REFERENCE.to_string() => {
             Ok(())
         }
-        None => {
-            super::mark_all_unapplied(gb_repository).map_err(errors::VerifyError::Other)?;
-            Err(errors::VerifyError::DetachedHead)
-        }
-        Some(head_name) => {
-            super::mark_all_unapplied(gb_repository).map_err(errors::VerifyError::Other)?;
```

Comment:
@schacon maybe you remember why we had this? anyways now it seems not needed

Reply:
@krlvi anyways, this is why it was unapplying. i think we've added it to avoid conflicts when switching to and from git, but since now we handle that by other means, should be safe to remove.

Reply:
I don't remember this. It looks like you introduced it here:  490318e8ea9415a66b2f3d0e17d48f41be7c96a8

Reply:
https://github.com/gitbutlerapp/gitbutler/pull/991